docs/design-proposals/multi-tenancy/multi-tenancy-network.md

```diff
@@ -292,3 +322,12 @@ Components not decided yet at current phase are not included in these views.
 ### Key Scenario: tenant & DNS provisioning
 
 ![tenant & DNS provision](images/network-scenario-tenant-init.png)
+
+### Activity Flow: Pod - from creation to running
+
+![Pod-activity-diag](images/pod-activity-diagram.png)
+
+1. It is strongly recommended to annotate new pods with arktos.futurewei.com/network-readiness as hint for scheduler. It is not mandatory, though - kubelet and cni plugin will have to cope with it and gets it right eventually. This kind of initial pod annotation can be done by a custom admission control.
```

Comment:
If the network-readiness annotation is not mandatory, how does the node agent know when it's ready to call CNI agents?

Reply:
kubelet will try to start pod after the pod is scheduled to the node mostly as current process, with addition of ensuring network status is ready associated to the pod. In case pod cannot gets its nic setup successfully, kubelet will try it later.
Absence of network-readiness is allowed for backward compatibility, particularly for flat network using regular cni plugins. Ideally pod with presence of such makes the workflow more efficient, but w/o it pod is still able to start, eventually. However, if a certain network cannot work in that relaxed way, then such network-readiness annotation is required for that type of network - in that case scheduler will ensure pod won't be scheduled until network operator has made necessary preparations.